packages/cli/src/commands/serve.ts

```diff
@@ -13,16 +15,45 @@ class PlatformAppDevelopmentServer {
 
     async serve(): Promise<void> {
         try {
-            const viteServer = await createServer({
+            const settingsSchema = await esbuild.context({
+                entryPoints: ['./src/index.ts'],
+                outfile: './dist/dev-settings.js',
+                minify: true,
+                globalName: 'devSettings',
+                format: 'iife',
+                bundle: true,
+                loader: {
+                    '.css': 'empty',
+                },
+            });
+
+            const viteServerDev = await createServer({
                 root: process.cwd(),
                 configFile: false,
-                plugins: [react()],
                 define: {
                     'process.env.NODE_ENV': JSON.stringify('development'),
                 },
+                plugins: [
+                    react(),
+                    {
+                        name: 'prebuild-commands',
+                        handleHotUpdate: ({ file, server }) => {
+                            const filePattern = globToRegExp('**/settings.ts');
+                            const filePattern2 = globToRegExp('**/index.ts');
```

Comment:
We have the `index.ts` path set as `entryFilePath` so we can remove the regex?
For the settings, do we set it in the manifest or is it always expected to be `src/settings.ts`?
In any case, I would remove the regex if possible to have the direct path, wdyt?

Reply:
What we were struggeling with was, that we needed an output of the bundle as IIFE to then include our Settings while in dev mode (not published). 

I didnt manage to have 2 builds in vite, one that serves the App and at the same time have the index.ts built as IIFE format under a specific path. Therefore i added a 2nd build with esBuild that outputs the 2nd build. It works well. 

But, now when esbuild generates a new file, the vite server doesnt serve the output of it. It doesnt know that smth changed in the file and doesnt pick up on it. Thats why i used the hotUpdate to restart with vite and have the file available. Now of course one can put their settings wherever they want, i just help them with restarting on index.ts (if they added their settings there) or when they have them in the settings.ts file. But a dev can put these wherever they want, then the mechanism wouldnt work and they would have to restart the dev server by hand. 

Im still hoping to find a way to do it with vite, but it seems not so straight forward.

Reply:
ah no no, I don't mind esbuild, just asking if we could have "direct path" rather than regex (`**/<filename>.ts`)?

```ts
handleHotUpdate: ({ file, server }) => {
    const indexFilePath = [...];
    const settinsgFilePath = [...];
    if (indexFilePath === file || settinsgFilePath === file) {
        // if the change is either in settings.ts or index.ts do a rebuild to load settings
        settingsSchema.rebuild();
        server.restart();
    }
},
```

Reply:
Else we would have a conflict if any other file is called `index.ts` or `settings.ts` (in nested folders)

Reply:
ahh sure, ill change it to that, your right 👍